pkg/apis/pipeline/v1alpha2/resource_types.go

```diff
@@ -96,6 +104,39 @@ type ResourceDeclaration struct {
 	TargetPath string `json:"targetPath,omitempty"`
 }
 
+// PipelineResourceBinding connects a reference to an instance of a PipelineResource
+// with a PipelineResource dependency that the Pipeline has declared
+type PipelineResourceBinding struct {
+	// Name is the name of the PipelineResource in the Pipeline's declaration
+	Name string `json:"name,omitempty"`
+	// ResourceRef is a reference to the instance of the actual PipelineResource
+	// that should be used
+	// +optional
+	ResourceRef *PipelineResourceRef `json:"resourceRef,omitempty"`
+
+	// FIXME(vdemeester) cannot support this if PipelineResource stays in alpha and on the same group
+	// ResourceSpec is specification of a resource that should be created and
+	// consumed by the task
+	// +optional
+	// ResourceSpec *PipelineResourceSpec `json:"resourceSpec,omitempty"`
```

Comment:
what if we move `resources` e.g. something like  `pkg/apis/resources/v1alpha1` <-- i think then we'd be able to import resources here?

Reply:
@bobcatfish I can see multiple ways indeed :angel: I just need to see how easy or hard it is to bend the generators :angel: :smiling_imp:

Reply:
I am going to try that change in a separate PR btw :angel:

Reply:
@bobcatfish experiment on this https://github.com/tektoncd/pipeline/pull/1773 :sweat_smile: